frame/executive/src/lib.rs

```diff
@@ -463,15 +463,16 @@ where
 			frame_system::InitKind::Inspection,
 		);
 
+		// Frame system only inserts the parent hash into the block hashes as normally we don't know
+		// the hash for the header before. However, here we are aware of the hash and we can add it
+		// as well.
+		frame_system::BlockHash::<System>::insert(header.number(), header.hash());
+
 		// Initialize logger, so the log messages are visible
 		// also when running WASM.
 		frame_support::debug::RuntimeLogger::init();
 
-		<AllModules as OffchainWorker<System::BlockNumber>>::offchain_worker(
-			// to maintain backward compatibility we call module offchain workers
-			// with parent block number.
```

Comment:
AFAIR it was because before that change we didn't initialize `frame_system` with new header data (the call was marked `skip_initialize`), so it was executed in the context of parent block.

I'm fine with the change, but this is definitely breaking, so we should make sure to include this in release notes.

Reply:
Ahh! Makes sense, but with the change to passing the header we already started executing on the "correct" block.